resource-validator/src/main/scala/com/broadinstitute/dsp/resourceValidator/DbReader.scala

```diff
@@ -36,6 +35,37 @@ object DbReader {
     sql"""select googleProject, initBucket from CLUSTER WHERE status="Deleted";"""
       .query[InitBucketToRemove]
 
+  // Return all non-deleted clusters with non-default nodepools that have apps that were all deleted
+  // or errored outside the grace period (1 hour)
+  //
+  // We are excluding clusters with only default nodepools running on them so we do not remove batch-pre-created clusters.
+  // We are calculating the grace period for cluster deletion assuming that the following are valid proxies for an app's last activity:
+  //    1. destroyedDate for deleted apps
+  //    2. dateAccessed for error'ed apps
+  // TODO: Read the grace period (hardcoded to '1 HOUR' below) from config
+  val kubernetesClustersToDeleteQuery =
+    sql"""
+            SELECT kc.id, kc.googleProject
+            FROM KUBERNETES_CLUSTER kc
+            WHERE
+              kc.status != "DELETED" AND
+              NOT EXISTS (
+                SELECT *
+                FROM NODEPOOL np
+                LEFT JOIN APP a ON np.id = a.nodepoolId
+                WHERE
+                  kc.id = np.clusterId AND np.isDefault = 0 AND
+                  (
+                    (a.status != "DELETED" AND a.status != "ERROR") OR
+                    (a.status = "DELETED" AND a.destroyedDate > now() - INTERVAL 1 HOUR) OR
+                    (a.status = "ERROR" AND a.dateAccessed > now() - INTERVAL 1 HOUR) OR
```

Comment:
wouldn't `a.status = "ERROR"` alone suffice? (without checkint `dateAccessed`)

Reply:
The `dateAccessed` check gives a grace period I think -- so we won't delete the cluster if an Error'd app is <1 hour old. 

I think I like this behavior, as a user might want to quickly recreate if they get a creation error.

Reply:
> we won't delete the cluster if an Error'd app is <1 hour old

If we're trying to check if the cluster is < 1hr old, shouldn't we be checking `createdDate`? still not sure why `dateAccessed` is needed

Reply:
Not the cluster but the app. Imagine a scenario where a user attempts to create an app but it errors out. In that case, we don't want to remove the cluster right away and instead, give the user some grace period in case they want to delete their app, create another app, etc.

Reply:
Hmmm...on another look, we don't seem to update `app.dateAccessed` upon app creation erroring out. In that case this query would be effectively checking `createdDate`.